src/sentry/utils/event_frames.py

```diff
@@ -32,51 +48,48 @@ class SdkFrameMunger:
     supported_sdks: Set[str] = field(default_factory=set)
 
 
-def java_frame_munger(key: str, frame: MutableMapping[str, Any]) -> bool:
-    if frame.get("filename") is None or frame.get("module") is None:
-        return False
-    if "/" not in str(frame.get("filename")) and frame.get("module"):
+def java_frame_munger(frame: EventFrame) -> str | None:
+    if not frame.filename or not frame.module:
+        return None
+    if "/" not in str(frame.filename) and frame.module:
         # Replace the last module segment with the filename, as the
         # terminal element in a module path is the class
-        module = frame["module"].split(".")
-        module[-1] = frame["filename"]
-        frame[key] = "/".join(module)
-        return True
-    return False
+        module = frame.module.split(".")
+        module[-1] = frame.filename
+        return "/".join(module)
+    return None
 
 
-def cocoa_frame_munger(key: str, frame: MutableMapping[str, Any]) -> bool:
-    if not frame.get("package") or not frame.get("abs_path"):
-        return False
+def cocoa_frame_munger(frame: EventFrame) -> str | None:
+    if not frame.package or not frame.abs_path:
+        return None
 
-    rel_path = package_relative_path(frame.get("abs_path"), frame.get("package"))
+    rel_path = package_relative_path(frame.abs_path, frame.package)
     if rel_path:
-        frame[key] = rel_path
-        return True
-    return False
+        return rel_path
+    return None
 
 
-def flutter_frame_munger(key: str, frame: MutableMapping[str, Any]) -> bool:
-    if not frame.get("abs_path"):
-        return False
+def flutter_frame_munger(frame: EventFrame) -> str | None:
```

Comment:
I modified each munger function to use `EventFrame` and return `str | None`, but otherwise kept the behavior the same.